pkg/maps/ctmap/types.go

```diff
@@ -104,6 +108,356 @@ func (m MapType) isTCP() bool {
 	return false
 }
 
+type CtKey interface {
+	bpf.MapKey
+
+	// ToNetwork converts fields to network byte order.
+	ToNetwork() CtKey
+
+	// ToHost converts fields to host byte order.
+	ToHost() CtKey
+
+	// Dump contents of key to buffer. Returns true if successful.
+	Dump(buffer *bytes.Buffer, reverse bool) bool
+
+	// GetFlags flags containing the direction of the CtKey.
+	GetFlags() uint8
+}
+
+// CtKey4 is needed to provide CtEntry type to Lookup values
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type CtKey4 struct {
+	DestAddr   types.IPv4      `align:"daddr"`
+	SourceAddr types.IPv4      `align:"saddr"`
+	DestPort   uint16          `align:"dport"`
+	SourcePort uint16          `align:"sport"`
+	NextHeader u8proto.U8proto `align:"nexthdr"`
+	Flags      uint8           `align:"flags"`
+}
+
+// NewValue creates a new bpf.MapValue.
+func (k *CtKey4) NewValue() bpf.MapValue { return &CtEntry{} }
```

Comment:
Could we make `NewValue` be part of the interface to make it clear that CT and NAT map have to override this, or technically not possible? Iirc, with the TupleKey back then, is was that both derived their types from tuple key and hence both had separate `NewValue` helpers. Would we otherwise need to cast incompatible value types (CT vs NAT value if default is the former)?